src/algorithms/blocked_gibbs_algorithm.cc

```diff
@@ -14,9 +18,36 @@ void BlockedGibbsAlgorithm::print_startup_message() const {
 }
 
 void BlockedGibbsAlgorithm::sample_allocations() {
-  return;  // TODO
+  auto &rng = bayesmix::Rng::Instance().get();
+  for (int i = 0; i < data.rows(); i++) {
+    // Compute weights
+    Eigen::VectorXd logprobas =
+        cond_mixing->get_weights(true, false, mix_covariates.row(i));
+    for (int j = 0; j < logprobas.size(); j++) {  // TODO?
```

Comment:
We don't really have a universal getter for the number of components in conditional mixings, so I basically used `get_weights().size()` to get it. I feel like this is kind of a cheat, or even better, a patch for a wider issue. Does the number of components belong to the prior in all conditional algorithms/mixings? If so, we have a useless `init_num_clusters` member hanging in the base algorithm but is only used for marginal algorithm. We should then move it to the Marginal class. Or even better, somehow create a ConditionalMixing getter method for the number of components. Extracting it from a proto would require a convention for conditional priors, which is not good. Thus we have to find another way. In the latter case, we would just need to set `init_num_clusters` to that value.

Reply:
No there are conditional algorithms for which init_num_clusters makes perfect sense!
In Conditional algorithms the number of components can be either fixed or random, so the way you recover it is fine.